### ISSUE TITLE
Web server migration

### ISSUE THREAD

**Issue by SubJunk** [UniversalMediaServer/UniversalMediaServer#1844]
Finally the web server is fully migrated. The machine is more powerful and also uses PHP 7 (up from PHP 5 on the old one).
These are big changes and there may have been mistakes so please let me know if you see anything. I have updated some of the PHP to work with v7 but may have missed some.

**Comment by begleysm** [comment]
I am unable to log into the Forums.  I was able to successfully get a new password via the "I forgot my password" button but, after activiating, that password doesn't work either.  I always get an error saying "The submitted form was invalid. Try submitting again."  I'm not sure if this related but it might be.  If you don't think it is, I can open a new ticket.

**Comment by SubJunk** [comment]
@begleysm I just tested and successfully signed up with an account. Can you try again?

**Comment by begleysm** [comment]
I tried again with no luck.  See below for my process.

I started by trying to login with my typical password
![image](https://user-images.githubusercontent.com/1275999/65892100-b6da1300-e373-11e9-82b4-013d772a493b.png)

I was met with this error message
![image](https://user-images.githubusercontent.com/1275999/65892128-c194a800-e373-11e9-96ed-b74d6abe3c24.png)

I went through the "I forgot my password" process and was emailed a new password (w/ activation link)
![image](https://user-images.githubusercontent.com/1275999/65892158-d113f100-e373-11e9-8564-77d4d2573c13.png)

I clicked the activation link (and even clicked it a second time today, getting the following message, just to make sure I hadn't forgotten)
![image](https://user-images.githubusercontent.com/1275999/65892228-ea1ca200-e373-11e9-869e-d45b46060fc7.png)

I tried logging in with the new password from the email and got the same negative result
![image](https://user-images.githubusercontent.com/1275999/65892253-f43ea080-e373-11e9-8f6c-321b0791dd64.png)

To be extra double sure I went through the "I forgot my password" process a second time... this time I got a super crazy long password (as shown below).  But it still didn't work after activating and trying this new crazy long password.
![image](https://user-images.githubusercontent.com/1275999/65892527-74fd9c80-e374-11e9-889d-b4b6680f509e.png)
![image](https://user-images.githubusercontent.com/1275999/65892542-7af37d80-e374-11e9-9736-f191015a3433.png)

**Comment by begleysm** [comment]
I had kinda forgotten about this and went to login on my phone (last time was from a Linux laptop) and again, couldn't login, reset my password, activated my account and attempted to login with the new, giant random, password.

Once again I got the "Submitted form was invalid.  Try submitting again."  error.

![Capture+_2019-10-31-05-57-03](https://user-images.githubusercontent.com/1275999/67948734-933cff00-fba3-11e9-822f-07db49dd177b.png)

**Comment by SubJunk** [comment]
@begleysm I'm not sure what it could be, maybe there is a maximum password length or something. Can you try a more simple one as a test? Also it's weird that the forum is sending your password in an email, I'll attempt to disable that, I'm amazed they do this by default

**Comment by Nadahar** [comment]
@SubJunk I've had similar problems logging in to the forum several times since you demoted me. I never had these problems when I had "developer status", so I suspect that permissions are involved in some mystical way - together with cookies I suspect.

I don't remember what I did to suddenly be able to log in again, but I think deleting the cookies should be the first step. At one time, I could log in from one of my computers, but not from another. I know that my password was correct, because I looked it up in the "password cache" on the computer that could log in.

I used Firefox for all of this, in case that matters. You should probably create a test user without special permissions, and then use that from a separate Firefox profile than your "normal" profile, if you use Firefox at all.

The only thing I know for sure, is that the password itself was never the problem. It has happened for me on several occasions, although it has been quite a while since I saw this now.

**Comment by SubJunk** [comment]
I see there is an available update to the forum software we use which has modified login forms for security so I'll update to that.
We are using a mostly unmodified version of the forum so it may be a bug with that software.

I have been testing on a regular user account by the way, I registered and logged in and out.

**Comment by SubJunk** [comment]
I have updated the forum software now. Not sure if that will fix it but it's worth a try

**Comment by begleysm** [comment]
@SubJunk I'm shocked but... the update worked!  I clicked "I forgot my password", put in my email, got a new super-long random password, and was able to login with it.  I was the able to change my password, logout, and the successfully login with the new password that I created.

Thanks for circling back around to this and not letting it rot forever.

@Nadahar thanks for confirming I'm not a lunatic who doesn't know how to use a computer haha.
